stargate/crates/stargate-lib/src/lib.rs

```diff
@@ -16,16 +17,23 @@ mod utilities;
 pub struct Stargate<'app> {
     service_list: HashMap<String, ServiceDefinition>,
     pub planner: QueryPlanner<'app>,
+    pub propagate_headers: Option<Vec<String>>,
+}
+
+pub struct StargateOpts<'app> {
+    pub schema: &'app str,
+    pub propagate_headers: Option<Vec<String>>,
 }
```

Comment:
Any specific reason you chose to model things like this?

I had imagined something like
```rust
struct Stargate<'app> {
  service_list: HashMap<String, ServiceDefinition>,
  pub planner: QueryPlanner<'app>,
  options: StargateOptions
}

struct StargateOptions {
  pub propagate_headers:Vec<String>
}

impl Default for StargateOptions {
  pub fn default() -> Self {
    Self {
      propagate_headers: vec![]
    }
  }
}
```

I don't think we need the schema in StargateOptions, it's already in the planner.
i.e. `StargateOptions` is not `Opts`

Also, in cases where the behavior for `None` is the same as `Some(vec![])`, I'd prefer making the type just a vector that can be empty.